lib/flapjack/data/migration.rb

```diff
@@ -125,6 +125,42 @@ def self.migrate_entity_check_data_if_required(options = {})
         end
       end
 
+      def self.clear_orphaned_entity_ids(options = {})
+        raise "Redis connection not set" unless redis = options[:redis]
+
+        logger = options[:logger]
+
+        semaphore = obtain_semaphore(ENTITY_DATA_MIGRATION, :redis => redis)
+        if semaphore.nil?
+          unless logger.nil?
+            logger.fatal "Could not obtain lock for data migration (entity id creation). Ensure that " +
```

Comment:
Would be good to s/entity id creation/clear orphaned entity ids/ in the logger message